### PR TITLE
reset top/left on resetFit

### LINKED ISSUE #29: max-width and max-height positioning issues
When setting `max-width` or `max-height` to the element, the first time positioning is properly calculated, but on the following `refit()` calls it will be positioned in the wrong place.
```html
<style>
  .sized {
    max-width: 200px;
  }
</style>

<simple-fit auto-fit-on-attach class="sized" onclick="this.refit()">
  Hello
</simple-fit>

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Fixes #29 by resetting top/left in `resetFit`

### CODE REVIEW

**Comment by valdrinkoshi** [comment]
@cdata kindly review :)

File: iron-fit-behavior.html [hunk 1]
```diff
@@ -171,14 +171,14 @@
...
-      }
       if (!this._fitInfo || !this._fitInfo.sizedBy.width) {
         this.sizingTarget.style.maxWidth = '';
-        this.style.left = this._fitInfo ? this._fitInfo.inlineStyle.left : '';
       }
+      if (!this._fitInfo || !this._fitInfo.sizedBy.height) {
+        this.sizingTarget.style.maxHeight = '';
+      }
+      this.style.top = this._fitInfo ? this._fitInfo.inlineStyle.top : '';
+      this.style.left = this._fitInfo ? this._fitInfo.inlineStyle.left : '';
```
Comment (cdata) [inline iron-fit-behavior.html]: IIRC there is a specific reason why you only want to do this if `!this._fitInfo.sizedBy.height / width` is true. Are you sure this change is safe?

Comment (valdrinkoshi) [inline iron-fit-behavior.html]: Here we are resetting the memoized data.
`this._fitInfo.inlineStyle` will contain the original `top, left`, and is independent from `this._fitInfo.sizedBy` which should be used for resetting `maxWidth, maxHeight`.
By associating them together, what happens is the following:
- on the first `refit()`, `left, top` are not set, hence we can calculate the proper position and set those values (memoized data)
- on the second `refit()`, `left, top` will not be reset, so the positioning will stay as is, with the current `top, left` (which is the memoized data, not the original `top, left`

I believe this is a safe change.

Comment (valdrinkoshi) [inline iron-fit-behavior.html]: @cdata FYI

**Comment by cdata** [comment]
LGTM :+1:

**Pull request merged**